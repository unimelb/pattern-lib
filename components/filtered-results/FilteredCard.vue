<template>
  <a
    href="/"
    :class="['filtered-results__card', getCategoryClassName()]">
    <span class="filtered-results__card-label">{{ item.category }}</span>
    <span class="filtered-results__card-category">Category name</span>
  </a>
</template>

<script>
export default {
  props: {
    item: {
      type: Object,
      required: true,
    },
  },
  methods: {
    getCategoryClassName() {
      return this.item.category === 'major' ? 'filtered-results__card--major' : 'filtered-results__card--minor';
    },
  },
};
</script>
