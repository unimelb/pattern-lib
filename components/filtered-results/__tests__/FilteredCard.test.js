import { mount, shallow } from 'vue-test-utils';

import FilteredCard from '../FilteredCard.vue';

describe('FilteredCard', () => {
  it('Should match snapshot', () => {
    const result = mount(FilteredCard,
      {
        propsData: {
          item: {
            type: 'major',
            name: 'Some category',
          },
        },
      }).element;
    expect(result).toMatchSnapshot();
  });

  it('Should have default prop and correct type', () => {
    const result = shallow(FilteredCard);
    const {
      item,
    } = result.vm.$options.props;
    expect(item.type).toBe(Object);
  });
});
