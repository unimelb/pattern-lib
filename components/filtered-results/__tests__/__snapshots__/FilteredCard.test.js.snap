// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilteredCard Should match snapshot 1`] = `
<a
  class="filtered-results__card filtered-results__card--major"
>
  <span
    class="filtered-results__card-label"
  >
    major
  </span>
   
  <span
    class="filtered-results__card-category"
  >
    Some category
  </span>
</a>
`;
